packages/app/src/cli/services/app-logs/ui/components/Logs.tsx

```diff
@@ -0,0 +1,207 @@
+import {AppEventData, subscribeProcess} from '../../../../services/logs.js'
+import {DeveloperPlatformClient} from '../../../../utilities/developer-platform-client.js'
+import React, {FunctionComponent, useRef, useState, useEffect} from 'react'
+
+import {Static, Box, Text} from '@shopify/cli-kit/node/ink'
+
+export interface LogsProps {
+  logsProcess: ({
+    jwtToken,
+    cursor,
+    filters,
+  }: {
+    jwtToken: string
+    cursor?: string | undefined
+    filters?: {
+      status?: string
+      source?: string
+    }
+  }) => Promise<{
+    cursor?: string | undefined
+    errors?: string[] | undefined
+    appLogs?: AppEventData[] | undefined
+  }>
+  developerPlatformClient: DeveloperPlatformClient
+  storeId: string
+  apiKey: string
+  cursor: string
+  jwtToken: string
+  filters?: {
+    status?: string
+    source?: string
+  }
+}
+
+interface DetailsFunctionRunLogEvent {
+  input: string
+  inputBytes: number
+  invocationId: string
+  output: string
+  outputBytes: number
+  logs: string
+  functionId: string
+  fuelConsumed: string
+  errorMessage: string | null
```

Comment:
Heads up- The input, output, and errorMessage types are going to change with this PR: https://github.com/Shopify/shopify/pull/513906